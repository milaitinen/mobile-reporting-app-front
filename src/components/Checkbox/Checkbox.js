import { CheckBox } from 'react-native-elements';
import React, { Component } from 'react';
import styles from './checkboxStyles';

class Checkbox extends Component {
    constructor(props) {
        super(props);
        this.state = {
            checked: props.isChecked || false,
        };
    }

    componentDidMount() {
        this.setState({ checked : this.props.defaultValue });
    }

    toggle = () => {
        this.setState({ checked: !this.state.checked });
    };

    render(){
        const editable = this.props.editable;
        const isPreview = this.props.isPreview;
        const unchecked = (editable || !isPreview) ? styles.$inactiveBlue : styles.$disabledGray;
        const text = (editable || !isPreview) ? styles.text : styles.disabledText;
        const container = (editable || !isPreview) ? styles.container : styles.disabled;

        return (
            <CheckBox
                title={this.props.title}
                checked={this.state.checked}
                disabled={!editable}
                iconType={'material'}
                uncheckedIcon={'check-box-outline-blank'}
                checkedIcon={'check-box'}
                uncheckedColor={unchecked}
                checkedColor={styles.$activeBlue}
                textStyle={this.state.checked ? styles.selectedText : text}
                containerStyle={this.state.checked ? styles.selectedContainer : container}
                onPress={() => {
                    this.props.onPressFunction((!this.state.checked) ? '1' : '0');
                    this.toggle();
                }}
            />
        );
    }
}



export default Checkbox;
