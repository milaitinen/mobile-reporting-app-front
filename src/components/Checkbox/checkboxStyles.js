import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    $inactiveBlue: '$inactive',
    $disabledGray: '$disabledBorder',
    $activeBlue: '$active',

    container: {
        backgroundColor: '$primaryWhite',
        height: 40,
        borderWidth: 0,
        marginLeft: 0,
        marginVertical: 0,
        marginBottom: 10,
        justifyContent: 'center',
    },

    selectedContainer: {
        backgroundColor: '$primaryWhite',
        height: 40,
        borderWidth: 2,
        borderColor: '$active',
        marginLeft: 0,
        marginVertical: 0,
        marginBottom: 10,
        justifyContent: 'center',
    },

    text: {
        color: '$placeholder',
        fontFamily: '$primaryFont',
        fontWeight: 'normal',
    },

    selectedText: {
        color: '$active',
        fontFamily: '$primaryFont',
        fontWeight: 'normal',
    },

    disabled: {
        backgroundColor: '$disabled',
        borderColor: '$disabledBorder',
        height: 40,
        borderWidth: 2,
        marginLeft: 0,
        marginVertical: 0,
        marginBottom: 10,
        justifyContent: 'center',
    },

    disabledText: {
        color: '$disabledPlaceholder',
        fontFamily: '$primaryFont',
        fontWeight: 'normal',
    }

});

export default styles;