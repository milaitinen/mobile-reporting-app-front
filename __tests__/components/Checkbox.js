import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Checkbox from '../../src/components/Checkbox/Checkbox';

configure({ adapter: new Adapter() });

describe('Checkbox tests', () => {
    it('renders correctly when not editable', () => {
        const tree = renderer.create(
            <Checkbox  title='some checkbox' editable={false} onPressFunction={jest.fn()}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when editable', () => {
        const tree = renderer.create(
            <Checkbox  title='some checkbox' editable={true} onPressFunction={jest.fn()}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls the onPressFunction when pressed', () => {
        const mockPress = jest.fn();
        const checkbox = shallow(
            <Checkbox title='some checkbox' isChecked={false} editable={true} onPressFunction={mockPress}/>
        );

        const checkBox = checkbox.find('CheckBox');
        checkBox.simulate('press');
        expect(mockPress).toHaveBeenCalled();
    });
});
