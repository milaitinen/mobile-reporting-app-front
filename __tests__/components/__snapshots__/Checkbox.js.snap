// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests renders correctly when editable 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        false,
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "#bfbfbf",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#43484d",
              "fontWeight": "bold",
              "marginLeft": 10,
              "marginRight": 10,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      some checkbox
    </Text>
  </View>
</View>
`;

exports[`Checkbox tests renders correctly when not editable 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        false,
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "#bfbfbf",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#43484d",
              "fontWeight": "bold",
              "marginLeft": 10,
              "marginRight": 10,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      some checkbox
    </Text>
  </View>
</View>
`;
